linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2934,7 +3307,7 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
         case Transient(NativeArrayWrapper(elemClass, nativeArray)) =>
           val newNativeArray = transformExprNoChar(nativeArray)
           elemClass match {
-            case ClassOf(elemTypeRef) =>
+            case ClassOf(elemTypeRef) if (elemTypeRef != LongRef) || useBigIntForLongs =>
```

Comment:
Note to self / TODO:
- continue here.
- understand why we need to exclude long here.